File: python/done/76.py
#!/usr/bin/python

mat=[]

for n in range(0,10):
 mat.append([])
 k=n
 print n
 while (k>=0):
  if(k==n):
   print "if"
   mat[n].append(1)
  else:
   print "else"
   mat[n].insert(0,(mat[n-k][k]+mat[n][k+1]))
  print k,"=",mat[n][k]
  k=k-1
 print


# here n,k     actually is  n,n-k
#      n,k = n-k,k + n,k+1    therefore =>  k,n-k + n,k-1

File: perl/76.pl
#!/usr/bin/perl

for($n=0;$n<=100000;$n++){
#  print $n,":\n"; 
  for($k=$n;$k>=0;$k--){
    if ($n eq $k) { $mat[$n][$k]=1; }
    else {
      $mat[$n][$k]=$mat[$n-$k][$k]+$mat[$n][$k+1];
    }
  }
  if ($n eq 100) {
    print $n,",",$mat[$n][1],"\n";
    last;
  }
#  print "\n";
}

